bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/admin/StoreAdminHandler.java

```diff
@@ -477,4 +490,85 @@ public Response getRemoteByUrl( final @PathParam( "packageType" ) String package
         return response;
     }
 
+
+    @ApiOperation( "Revalidation of Artifacts Stored on demand" )
+    @ApiResponses( { @ApiResponse( code = 200, response = ArtifactStore.class, message = "Revalidation for Remote Repositories was successfull" ),
+        @ApiResponse( code = 404, message = "Revalidation is not successfull" ), } )
+    @Path( "/revalidate" )
+    @POST
+    public Response revalidateArtifactStores(
+        @PathParam("package") String packageType,
+        @PathParam( "type" ) String type) {
+
+        ArtifactStoreValidateData result = null;
+        Map<String,String> results = new HashMap<>();
+        Response response;
+
+        try {
+            StoreType storeType =  StoreType.get(type);
+
+            List<ArtifactStore> allArtifactStores = adminController.getAllOfType(packageType, storeType);
```

Comment:
For consistency, let's put the StoreValidator into AdminController, then package up this logic into a new method in AdminController.

At this point it's not terribly important, but we have that separation in order to make it relatively cheap to switch over to a new service endpoint technology if we need to. We would have all the business logic in the controllers, and just have to wire up the new endpoint technology to those controllers.